components/admin_console/system_user_detail/team_list/abstract_list.tsx

```diff
@@ -2,79 +2,91 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
 
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
 import NextIcon from 'components/widgets/icons/fa_next_icon';
 import PreviousIcon from 'components/widgets/icons/fa_previous_icon';
 
 import './abstract_list.scss';
 
 const PAGE_SIZE = 10;
 
-export default class AbstractList extends React.PureComponent {
-    static propTypes = {
-        userId: PropTypes.string.isRequired,
-        headerLabels: PropTypes.array.isRequired,
-        data: PropTypes.arrayOf(PropTypes.object),
-        onPageChangedCallback: PropTypes.func,
-        total: PropTypes.number.isRequired,
-        renderRow: PropTypes.func.isRequired,
-        emptyListTextId: PropTypes.string.isRequired,
-        emptyListTextDefaultMessage: PropTypes.string.isRequired,
-        actions: PropTypes.shape({
-            getTeamsData: PropTypes.func.isRequired,
-            removeGroup: PropTypes.func,
-        }).isRequired,
+type Props = {
+    userId: string;
+    headerLabels: Record<string, any>[];
+    data: Record<string, any>[];
+    onPageChangedCallback?: (paging: Paging) => void;
+    total: number;
+    renderRow: (item: {[x: string]: string}) => JSX.Element;
+    emptyListTextId: string;
+    emptyListTextDefaultMessage: string;
+    actions: {
+        getTeamsData: (userId: string) => ActionFunc & Partial<{then: (func: () => void) => void}> | Promise<Record<string, any>>;
```

Comment:
Can't this just be `Promise<void>` since the function doesn't return anything and we only need `then()` from the `Promise`?

Reply:
Yes i tried that, but that produces an error on `team_list.tsx` because it passes a `getTeamsData` to `abstract_list.tsx` too and the types won't match then.

Reply:
Yes i tried that, but it produces an error because `team_list.jsx` also passes a `getTeamsData` to `abstract_list.tsx` and the type won't match

Reply:
What about just `Promise<Record<string, any>>`?

Reply:
That doesn't work either

Reply:
Weird. Alright we can stick with this for now.